server/afk.lua

```diff
@@ -1,9 +1,87 @@
 local QBCore = exports['qb-core']:GetCoreObject()
+local loggedInPlayers = {}
+local checkUser = {}
+local previousPos = {}
+local time = {}
+local timeMinutes = {
+    ['900'] = 'minutes',
+    ['600'] = 'minutes',
+    ['300'] = 'minutes',
+    ['150'] = 'minutes',
+    ['60'] = 'minutes',
+    ['30'] = 'seconds',
+    ['20'] = 'seconds',
+    ['10'] = 'seconds'
+}
 
-RegisterNetEvent('KickForAFK', function()
-	DropPlayer(source, 'You Have Been Kicked For Being AFK')
+local function updateCheckUser(source)
+    local permissions = QBCore.Functions.GetPermission(source)
+
+    for k in pairs(permissions) do
+        if Config.IgnoreGroupsForAFK[k] then
+            checkUser[source] = false
+            return
+        end
+    end
+
+    checkUser[source] = true
+end
+
+RegisterNetEvent('QBCore:Server:OnPlayerLoaded', function()
+    loggedInPlayers[source] = true
+    updateCheckUser(source)
 end)
 
-QBCore.Functions.CreateCallback('qb-afkkick:server:GetPermissions', function(source, cb)
-    cb(QBCore.Functions.GetPermission(source))
+AddEventHandler('QBCore:Server:OnPlayerUnload', function(source)
+    loggedInPlayers[source] = false
+    checkUser[source] = false
+end)
+
+AddEventHandler('QBCore:Server:OnPermissionUpdate', function(source)
+    updateCheckUser(source)
+end)
+
+CreateThread(function()
+    for _, v in pairs(GetPlayers()) do
+        v = tonumber(v)
+        loggedInPlayers[v] = Player(v).state.isLoggedIn
+        if loggedInPlayers[v] then
+            updateCheckPlayer(v)
```

Comment:
```suggestion
            updateCheckUser(v)
```
Whoops, should be `updateCheckUser`, not `updateCheckPlayer`.
Player does indeed sound better though